gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -609,15 +620,17 @@ private int peekKeyword() throws IOException {
     String keyword;
     String keywordUpper;
     int peeking;
-    if (c == 't' || c == 'T') {
+
+    // Uppercase letters are not recognized if strict mode is used.
+    if (c == 't' || (!strict && c == 'T')) {
```

Comment:
Rather than checking `strict` in all these ifs, I think it would be simpler to start the `for` loop below at 0 instead of 1. Also, instead of checking `strict` in the loop, before the loop you could set `keywordUpper` to be the same as `keyword` if `strict` is true. (Maybe renaming it to something like `alternativeKeyword`.)

Reply:
I've rewritten this in the most recent version. Personally, I'm not convinced its cleaner than what was there before so let me know what you think.

Reply:
@eamonnmcmanus What do you think of this bit of logic has been rewritten?